draft-ietf-lamps-csr-attestation.md

```diff
@@ -527,34 +527,45 @@ EvidenceStatement ::= SEQUENCE {
-select an appropriate Verifier (software) without the need to perform any
-parsing of the `EvidenceStatement.stmt` data.
-To help with this, the Attester MAY populate the hint with a name of a
-software package that will be capable of parsing this data.
-The hint SHOULD contain a value that is unique
-to this Verifier, for example,  a fully qualified domain name (FQDN), a uniform
-resource name (URN) {{RFC8141}}, or a registered value corresponding to this
-Evidence format.
-In a typical usage scenario, the RP is pre-configured with a list of trusted
-Verifiers and the corresponding hint values can be used to look up appropriate Verifier.
-Tricking an RP into interacting with an unknown and untrusted Verifier has to be avoided
-as the retrieved Attestation Result must be reliable.
+Based on the responsibilities of the different roles in the RATS architecture,
+Relying Parties need to relay Evidence to Verifiers for evaluation and obtain
+an Attestation Result in return. Ideally, the Relying Party should select a Verifier
+based on the received Evidence without requiring the Relying Party to inspect the
+Evidence itself. This "routing" decision is simple when there is only a single
+Verifier configured for use by a Relying Party but gets more complex when there
+are different Verifiers available and each of them capable of parsing only certain
+Evidence formats.
+
+In some cases, the EvidenceStatement.type OID will be sufficient information
+for the Relying Party to correctly route it to an appropriate Verifier,
+however since the type OID only identifies the general data format, it is possible
+that multiple Verifiers are registered against the same type OID in which case the
+Relying Party will either require additional parsing of the evidence statement, or
+the Attester will be required to provide additional information.
+
+To simplify the task for the Relying Party an optional field, the hint, is available
+in the EvidenceStatement structure, as shown in {{code-EvidenceStatement}}. An
+Attester MAY include the hint to the EvidenceStatement and it MAY be processed
+by the Relying Party. The Relying Party MAY decide not to trust the information
+embedded in the hint or policy MAY override any information provided by the Attester
+via this hint.
+
+When the Attester populates the hint, it MUST contain a fully qualified domain
+name (FQDN) which uniquely identifies a Verifier.
+The problem of mapping hint FQDNs to Verifiers, and the problem of FQDN collision
+is out of scope for this specification; it is assumed that Attester and Verifier
+makers can manage this appropriately on their own FQDN trees, however if this
+becomes problematic then a public registry may be needed.
+
+In a typical usage scenario, the Relying Party is pre-configured with
+a list of trusted Verifiers and the corresponding hint values can be used to look
+up appropriate Verifier. Tricking an Relying Party into interacting with an unknown
```

Comment:
This sentence needs to be strengthened to say that the FQDN is simply an identifier. It MUST NOT be dereferenced, looked up in DNS, or FETCHED.

Question: is "FQDN" misleading here? Would calling it a URI make it less dangerous (or more dangerous??).

Reply:
> Question: is "FQDN" misleading here? Would calling it a URI make it less dangerous (or more dangerous??).

Per this topic of discussion in an editors' meeting, this is specifically why I mentioned URN is preferable to URI. There is a reason I believed that was picked in the version modified by this PR.

Reply:
> > Question: is "FQDN" misleading here? Would calling it a URI make it less dangerous (or more dangerous??).
> 
> Per this topic of discussion in an editors' meeting, this is specifically why I mentioned URN is preferable to URI. There is a reason I believed that was picked in the version modified by this PR.

Are you saying URNs aren't dereferenced, looked up in DNS or FETCHED? I would think just the opposite is more likely.

Reply:
> Are you saying URNs aren't dereferenced, looked up in DNS or FETCHED? I would think just the opposite is more likely.

I was trying to avoid comment on de-referencing for now, if you are referring to reply in response to Mike's question. A URN cannot be de-referenced in the same way, and the previous section that this PR changes originaly referenced URN and RFC8141.

By using a name that cannot refer explicitly to a location (if I can be casual with the definition), it furthers the allowlist concept ("the Relying Party is pre-configured with
a list of trusted Verifiers and the corresponding hint values can be used to look
up appropriate Verifier") and conceptually limits the misuse and usage of DNS _without_ allowlist, for those who enjoy ignoring the recommendations of the spec, as an added benefit.